File: mbta-with-friends/src/models/friend.ts
export interface IFriend {
  id: number;
  username: string;
  is_friend: boolean;
}

export interface NFriend {
  username: string;
}

export class Friend implements IFriend {
  public id: number;
  public username: string;
  public is_friend: boolean;

  constructor(friend: IFriend) {
    this.id = friend.id;
    this.username = friend.username;
    this.is_friend = friend.is_friend;
  }
}

File: src/models/friend.rs
use crate::schema::friends;

#[derive(Queryable, QueryableByName, Serialize)]
#[table_name="friends"]
pub struct Friend {
    pub id: i32,
    pub user_id: i32,
    pub friend_id: i32,
}

#[derive(Insertable)]
#[table_name="friends"]
pub struct NewFriend<'a> {
    pub user_id: &'a i32,
    pub friend_id: &'a i32,
}

#[derive(Deserialize)]
pub struct NFriend {
    pub username: String,
}
